api/namex/utils/common.py

```diff
@@ -96,6 +99,24 @@ def convert_to_ascii(value):
         return value
 
 
+def convert_to_utc_min_date_time(date_str: str):
+    date_obj = datetime.strptime(date_str, '%Y-%m-%d').date()
+    min_time = time(hour=0, minute=0, second=0, microsecond=0)
+    return datetime.combine(date_obj, min_time, tzinfo=tz.UTC)
+
+
+def convert_to_utc_max_date_time(date_str: str):
+    date_obj = datetime.strptime(date_str, '%Y-%m-%d').date()
+    max_time = time(hour=23, minute=59, second=59, microsecond=999999)
+    return datetime.combine(date_obj, max_time, tzinfo=tz.UTC)
+
+
+# def convert_to_utc_date_time(date_time_str: str):
+#     """Convert datetime string with utc offset to datetime object"""
+#     date_time_obj = datetime.datetime.strptime(date_time_str, '%Y-%m-%d %H:%M:%S%z')
+#     return date_time_obj.astimezone(UTC)
```

Comment:
Is this commented out because it's not used? What's the normal style here: leave in commented-out code, or delete it?

Reply:
> > This the current implementation you bet. I think Severins concern is when the results are large (in this case over 1000), the request takes some time to resolve and he would prefer a quick response with count prior to a response with all the results.
> > @severinbeauvais Correct me if i'm wrong here tho!
> 
> My concern is that we might pass a huge amount of data to the customer and then just throw it away. Some customers don't have a fast network and some customers pay a lot for their bandwidth, so this is a bad design.
> 
> Maybe this needs to be pushed back to Scott, ie, we would display the first 1000 anyway since we went through the trouble to find them and return them.
> 
> _Updated:_
> A fast response when there are too many results and you don't display anything is a nice side effect of this, yes.

Sorry I didn't see all these comments earlier. The API will only return the amount specified. So if you only want it to display 10 NRs you can do that by specifying rows=10 in the params. It will give you the total amount that matched in the reesponse (so you could show 10 out of 1 000,000 to the customer). It also supports paging with the 'start' param if that is also needed for you (so you could get 10 NRs starting from the 50th row for example)

Reply:
Removed.